### PR TITLE
Temporal adapted conditional effect remover

### PR DESCRIPTION
**Pull request by Framba-Luca**

### CODE REVIEW

File: upf/conditional_effects_remover.py [hunk 1]
```diff
@@ -136,24 +208,47 @@ def _create_problem_copy(self):
             new_problem.add_object(o)
         for fl, v in self._problem.initial_values().items():
             new_problem.set_initial_value(fl, v)
+        for t, el in self._problem.timed_effects():
+            for e in el:
+                if e.is_conditional():
+                    raise # NOTE is it a problem if a timed_effect is conditional? It should be if this class is used!
+                new_problem.add_timed_effect(t, e)
```
[the remark this answers is not in the document]
Comment (mikand) [reply upf/conditional_effects_remover.py:215]: Good point. I would go for the third option: raise an exception stating that it is not possible to remove all the conditional effects.

If we want to be precise, we could remove the conditional effects for boolean variables as follows:
```
when psi then v  := phi
```
can be converted into 
```
v := (psi and phi) or ((not psi) and v)
```

**Review by mikand — CHANGES_REQUESTED** [review]
There are some opportunities for refactoring, but it looks almost perfect! Good job @Framba-Luca

File: upf/conditional_effects_remover.py [hunk 2]
```diff
@@ -126,6 +178,26 @@ def _shallow_copy_action_without_conditional_effects(self, action: Action) -> Ac
...
+        #emulates a do-while loop: searching for an available name
+        is_unavailable_name = True
+        while is_unavailable_name:
+            new_action_name = action.name()+ "_" +str(self._counter)
+            self._counter = self._counter + 1
+            is_unavailable_name = self._problem.has_action(new_action_name)
+        new_parameters = OrderedDict()
+        for ap in action.parameters():
+            new_parameters[ap.name()] = ap.type()
+        new_action = DurativeAction(new_action_name, new_parameters, self._env)
```
[the remark this answers is not in the document]
Comment (Framba-Luca) [reply upf/conditional_effects_remover.py:191]: Yes, definitely!

File: upf/transformers/conditional_effects_remover.py [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+                    if len(nda.effects()) > 0:
+                        if self._check_and_simplify_conditions(nda):
+                            self._action_mapping[nda] = action
+                            self._new_problem.add_action(nda)
+                            self._counter += 1
+            else:
+                raise NotImplementedError
+        return self._new_problem
+
+    def _check_and_simplify_conditions(self, action: DurativeAction) -> bool:
```
Comment (alvalentini) [inline upf/transformers/conditional_effects_remover.py:131]: @Framba-Luca Can these methods be moved to the `Remover` class?

**Pull request merged**